### PR TITLE
Chore: update deps

### PR DESCRIPTION
**Pull request by stephhuynh18**
## Description 

** Most significant changes in IPFS Daemon
Updating the dependencies of all the packages for https://github.com/ceramicnetwork/js-ceramic/issues/1804

Missed: 

## How Has This Been Tested?

-All tests are running
-Manually tested ipfs daemon

## PR checklist

Before submitting this PR, please make sure:

- [x] I have tagged the relevant reviewers and interested parties
- [x] I have updated the READMEs of affected packages
- [x] I have made corresponding changes to the documentation

## References:

https://github.com/ceramicnetwork/js-ceramic/issues/1804

### CODE REVIEW

File: packages/common/package.json [hunk 1]
```diff
@@ -32,24 +32,24 @@
   },
   "dependencies": {
     "@ceramicnetwork/streamid": "^1.3.5-rc.0",
-    "@overnightjs/logger": "^1.2.0",
     "caip": "~1.0.0",
-    "cids": "~1.1.6",
     "cross-fetch": "^3.1.4",
     "flat": "^5.0.2",
+    "jet-logger": "^1.0.6",
```
Comment (ukstv) [inline packages/common/package.json:38]: wow, cool

File: packages/common/src/anchor-service.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import type CID from 'cids'
+import { CID } from 'multiformats/cid'
```
Comment (ukstv) [inline packages/common/src/anchor-service.ts:1]:
```suggestion
import type { CID } from 'multiformats/cid'
```

File: packages/common/src/pinning.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import type CID from 'cids'
+import { CID } from 'multiformats/cid'
```
Comment (ukstv) [inline packages/common/src/pinning.ts:1]: import type

File: packages/common/src/stream.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import type CID from 'cids'
+import { CID } from 'multiformats/cid'
 import cloneDeep from 'lodash.clonedeep'
 import type { Context } from './context'
 import { StreamID } from '@ceramicnetwork/streamid'
```
Comment (ukstv) [inline packages/common/src/stream.ts:4]: import type

File: packages/core/src/__tests__/ceramic-pinning.test.ts [hunk 5]
```diff
@@ -141,12 +141,11 @@ describe('Ceramic stream pinning', () => {
       anchor: false,
       publish: false,
     })
-    await expect(isPinned(ceramic, stream.id)).resolves.toBeFalsy()
+    expect(isPinned(ceramic, stream.id)).resolves.toBeFalsy()
```
Comment (ukstv) [inline packages/core/src/__tests__/ceramic-pinning.test.ts:144]: why no `await` here? @stephhuynh18

File: packages/core/src/pubsub/pubsub.ts [hunk 6]
```diff
@@ -1,11 +1,11 @@
 import { Observable, EMPTY, pipe, of, from, Subscription, UnaryFunction } from 'rxjs'
-import { deserialize, KeepaliveMessage, MsgType, PubsubMessage, serialize } from './pubsub-message'
+import { deserialize, PubsubMessage, serialize } from './pubsub-message'
 import { IpfsApi } from '@ceramicnetwork/common'
 import { map, catchError, mergeMap, withLatestFrom } from 'rxjs/operators'
 import { IncomingChannel, filterExternal, IPFSPubsubMessage } from './incoming-channel'
 import { DiagnosticsLogger, ServiceLogger } from '@ceramicnetwork/common'
 import { TextDecoder } from 'util'
-import uint8ArrayToString from 'uint8arrays/to-string'
+import { toString as uint8ArrayToString } from 'uint8arrays'
```
Comment (ukstv) [inline packages/core/src/pubsub/pubsub.ts:8]: why this?

File: packages/ipfs-daemon/package.json [hunk 7]
```diff
@@ -32,21 +32,20 @@
...
-    "blockcodec-to-ipld-format": "^1.0.0",
-    "dag-jose": "^0.3.0",
-    "datastore-fs": "^5.0.1",
-    "datastore-level": "^6.0.1",
-    "datastore-s3": "^6.0.0",
+    "aws-sdk": "^2.1034.0",
+    "dag-jose": "^1.0.0",
+    "datastore-fs": "^6.0.1",
+    "datastore-level": "^7.0.1",
+    "datastore-s3": "^8.0.0",
```
Comment (ukstv) [inline packages/ipfs-daemon/package.json:39]: Ooh, this makes me nervous. Are you sure the major bump would not result in changed behaviour? Do you think their changelog is safe enough?

File: packages/core/src/__tests__/ceramic-pinning.test.ts (hunk 5 above)
Comment (stephhuynh18) [reply packages/core/src/__tests__/ceramic-pinning.test.ts:144]: I am reaching an odd bug where the tests will hang indefinitely on this test. Specifically the `isPinned()` on line 148. I removed the `awaits` and the tests behaved accordingly (ex. if i change `toBeFalsy()` to `toBeTruthy` the test fails, no changes means the tests pass). But looking at the documentation I am wrong. But why does it hang on `isPinned` on only this line but not on the other lines.... If you have any insights pls let me know

File: packages/core/src/pubsub/pubsub.ts (hunk 6 above)
Comment (stephhuynh18) [reply packages/core/src/pubsub/pubsub.ts:8]: If I `console.log(uint8ArrayToString)` with the original line, it logs `undefined`. I do not know why but it works if I export the other way. If you have any insights please let me know =)

File: packages/ipfs-daemon/package.json (hunk 7 above)
Comment (stephhuynh18) [reply packages/ipfs-daemon/package.json:39]: For `datastore-fs` and `datastore-level` the breaking change is `deep imports/requires are no longer possible`. 
For `datastore-s3` the two breaking changes are `only uses named exports` and `uses new interface-datastore types`. The new interface doesn't change much from what I can see. But I also am uncomfortable as we don't have any tests and I only did a few manual tests. I can change this back and do all of them or only `datastore-s3` in the future.

File: packages/core/src/pubsub/pubsub.ts (hunk 6 above)
Comment (ukstv) [reply packages/core/src/pubsub/pubsub.ts:8]: All right then. Something with module resolution: ESM vs CommonJS.

File: packages/ipfs-daemon/package.json (hunk 7 above)
Comment (ukstv) [reply packages/ipfs-daemon/package.json:39]: Okay, then we'll see how it behaves when deployed.

File: packages/core/src/__tests__/ceramic-pinning.test.ts (hunk 5 above)
Comment (stephhuynh18) [reply packages/core/src/__tests__/ceramic-pinning.test.ts:144]: @ukstv So i did some investigating and this is what I found:
`stream.update()`  --> `ceramic.applyCommit()` --> `repo.applyCommit()` 
It then calls
`stateManager.applyCommit()` --> `repo.load()` --> `repo.handlePinOpts()` --> `repo.unpin()`
After the above completes the following gets called 
`repo.applyWriteOpts()` --> `repo.handlePinOpts()` --> `repo.unpin()`

So essentially `repo.unpin()` gets called twice. This functions is async and so this double call may be whats confusing jest causing it to hang even after the test successfully completes . If I comment out one of `repo.handlePinOpts` calls the test passes and doesn't hang. If i don't unpin in the test, the test passes and doesn't hang.

I'm not sure if I am right so please help me! The duplicated call to unpin is unexpected

File: packages/key-did-resolver/src/__tests__/ed25519.test.ts [hunk 8]
```diff
@@ -1,12 +1,12 @@
...
-import multibase from 'multibase'
+import { base58btc } from 'multiformats/bases/base58'
 import * as mapper from '../ed25519'
 
 describe('Ed25519 mapper', () => {
   it('successfully resolves the document from did', async () => {
     const id = 'z6MktvqCyLxTsXUH1tUZncNdVeEZ7hNh7npPRbUU27GTrYb8'
 
-    const multicodecPubKey = multibase.decode(id)
+    const multicodecPubKey = base58btc.decode(id)
```
Comment (stephhuynh18) [inline packages/key-did-resolver/src/__tests__/ed25519.test.ts:9]: PLEASE CHECK

The old multibases used this https://github.com/multiformats/js-multibase/blob/master/src/constants.js to determine the encoding. 

This does not exist in multiformats. I am assuming its base58btc as the tests begin with a z. Not sure if this always be the case.

File: packages/key-did-resolver/src/index.ts [hunk 9]
```diff
@@ -34,7 +34,7 @@ export function getResolver(): ResolverRegistry {
         didDocumentMetadata: {},
       }
       try {
-        const multicodecPubKey: any = multibase.decode(parsed.id)
+        const multicodecPubKey: any = base58btc.decode(parsed.id)
```
Comment (stephhuynh18) [inline packages/key-did-resolver/src/index.ts:37]: PLEASE CHECK

The old multibases used this https://github.com/multiformats/js-multibase/blob/master/src/constants.js to determine the encoding.

This does not exist in multiformats. I am assuming its base58btc as in the tests the id begin with a z. Not sure if this always be the case.

File: packages/core/src/__tests__/ceramic-pinning.test.ts (hunk 5 above)
Comment (stephhuynh18) [reply packages/core/src/__tests__/ceramic-pinning.test.ts:144]: well i guess nvm the tests passed....

Comment (ukstv) [reply packages/core/src/__tests__/ceramic-pinning.test.ts:144]: @stephhuynh18 Your finding is true, so https://github.com/ceramicnetwork/js-ceramic/pull/1872

**Comment by ukstv** [comment]
Please, @stephhuynh18 direct this PR to `release/2.x-progress` branch.

**Review by ukstv — APPROVED** [review]

**Comment by stephhuynh18** [comment]
@ukstv not sure if we want to merge #1872 into this branch first before merging this branch into release/2.x-progress or merge this into the release/2.x-progress first and then merge #1872 after.

**Comment by ukstv** [comment]
I guess, we are going to merge https://github.com/ceramicnetwork/js-ceramic/pull/1872 into this, and merge this into `release/2.x-progress` branch.

**Comment by oed** [comment]
Any blockers here?

**Comment by ukstv** [comment]
@oed This one: https://github.com/ceramicnetwork/js-ceramic/pull/1872 pending review checkbox from @stbrody

**Comment by ukstv** [comment]
@stephhuynh18 Mind resolving the conflicts, please?

**Pull request merged**